src/parquet/schema.cc

```diff
@@ -720,17 +718,15 @@ int SchemaDescriptor::ColumnIndex(const std::string& node_path) const {
   return search->second;
 }
 
-int SchemaDescriptor::ColumnIndex(const Node& node) const {
-  int result = ColumnIndex(node.path()->ToDotString());
-  if (result < 0) {
-    return -1;
-  }
-  DCHECK(result < num_columns());
-  if (!node.Equals(Column(result)->schema_node().get())) {
-    // Same path but not the same node
-    return -1;
+int SchemaDescriptor::ColumnIndex(const Node* node) const {
```

Comment:
Why did this change from reference to pointer? We use references everywhere where the passed object cannot be null.

Reply:
Hmm... I didn't know this guideline (but out parameters are passed as pointers apparently?).
The search is done using pointer equality, so I thought passing a pointer would be more explicit (and perhaps less error-prone, in case the compiler does a temporary copy).

Reply:
By the way other methods such `Node::Equals` take a node pointer, even though passing a null pointer isn't supported. Should I still convert back to a reference?

Reply:
@wesm What is your opinion on this?

Reply:
Sorry for the delay on this. I think we should consistently use const-references (unless null is a valid input), comparing pointer equality with `&value` where necessary. See commentary in https://google.github.io/styleguide/cppguide.html#Reference_Arguments

Reply:
Ok, I've turned the argument back into a const-reference.